### PR TITLE
fix(executor, frontend): `StreamHopWindow` executor should derive window expr in frontend

### PR DESCRIPTION
**Pull request by jon-chuang**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?
Stream Hop executor should derive expr in frontend. e2e test previously panicked with variable duration interval due to not inlining session timezone.

Notes:
- wait for https://github.com/risingwavelabs/risingwave/pull/8403 to be merged

## Checklist For Contributors

- [X] I have written necessary rustdoc comments
- [X] I have added necessary unit tests and integration tests
- [X] I have added fuzzing tests or opened an issue to track them. (Optional, recommended for new SQL features #7934).
- [X] I have demonstrated that backward compatibility is not broken by breaking changes and created issues to track deprecated features to be removed in the future. (Please refer to the issue)
- [X] All checks passed in `./risedev check` (or alias, `./risedev c`)

## Documentation

- [X] My PR **DOES NOT** contain user-facing changes.

## Related
Fixes part of: https://github.com/risingwavelabs/risingwave/issues/7334

### CODE REVIEW

File: src/stream/src/executor/hop_window.rs [hunk 1]
```diff
@@ -280,28 +196,35 @@ mod tests {
...
-
         let input =
             MockSource::with_chunks(schema.clone(), pk_indices.clone(), vec![chunk]).boxed();
-
         let window_slide = IntervalUnit::from_minutes(15);
         let window_size = IntervalUnit::from_minutes(30);
-        let default_indices: Vec<_> = (0..5).collect();
-        let executor = super::HopWindowExecutor::new(
+        let (window_start_exprs, window_end_exprs) =
+            make_hop_window_expression(DataType::Timestamp, 2, window_size, window_slide).unwrap();
```
Comment (jon-chuang) [inline src/stream/src/executor/hop_window.rs:204]: at least the duplicated code is moved into a util

**Comment by liurenjie1024** [comment]
I think this may break existing pipeline.

**Comment by jon-chuang** [comment]
> I think this may break existing pipeline.

Hmm, why is that?

**Review by st1page — APPROVED** [review]
LGTM

**Comment by st1page** [comment]
> > I think this may break existing pipeline.
> 
> Hmm, why is that?

I think it is because we persist in the streaming plan proto in the meta store and our new implementation can not handle the old plan.

**Review by chenzl25 — APPROVED** [review]
LGTM. It is a breaking change and seems to be unable to provide backward compatibility.

**Pull request merged**